client/src/components/NearestHospital/HealthFacilities.js

```diff
@@ -151,28 +154,70 @@ const NearestHealthFacilities = props => {
 
   const onSearchInputChange = (value) => {
     setSearchValue(value);
-    console.log(searchValue);
-
   }
   const clearSearch = () => {
     setSearchValue('');
   }
 
-  const [userCoords, setCoords] = useState(null);
+  const [ userCoords, setCoords ] = useState(null);
+  const [ gettingLocation, setGettingLocation ] = useState(false);
 
   const userLocation = () => {
     if(navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition((position, error) => {
-        if(error) {
-          // change error message
-          alert("Sorry we could not get your location. Your browser may not be compatible")
+      setGettingLocation(true);
+
+        const errorCallback_highAccuracy = error => {
+            if (error.code === error.TIMEOUT) {
+              navigator.geolocation.getCurrentPosition(
+                 successCallback,
+                 errorCallback_lowAccuracy,
+                 {maximumAge:600000, timeout:30000, enableHighAccuracy: false}
+               );
+
+              return;
+            }
+            setGettingLocation(false);
+              let errorMsg;
+              if (error.code === 1) {
```

Comment:
good to see there are error codes, didn't know these were added to the backend
we could use this as a map from code / messages or use a switch case as well to return the right message... but here it's ok for now with the if / else because there are only a few codes